src/figuresClasses.ts

```diff
@@ -1,19 +1,83 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+function notTriangle(side1: number, side2: number, side3: number) :boolean {
+  const longest: number = Math.max(side1, side2, side3);
 
+  return (side1 + side2 + side3) - longest <= longest;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape: Shape;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
+
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('your error message');
+    }
+
+    if (notTriangle(a, b, c)) {
+      throw new Error('your error message');
+    }
+  }
+
+  getArea(): number {
+    const area: number = (this.a + this.b + this.c) / 2;
 
+    return Math.floor(Math.sqrt(area * (
+      area - this.a
+    ) * (area - this.b) * (area - this.c)) * 100) / 100;
```

Comment:
you round down to hundredths on each getArea, maybe better create a separate method for it?